lib/generator.ml

```diff
@@ -154,44 +185,46 @@ let rec process_schema_type ~ancestors (schema : schema) =
   | Some _, _ -> failwith "only string enums are supported"
   | None, _ ->
   match schema.typ with
-  | Some Integer -> maybe_nullable (process_int_type schema)
+  | Some Integer -> maybe_nullable (process_int_type state schema)
   | Some Number -> maybe_nullable "float"
   | Some String -> maybe_nullable "string"
   | Some Boolean -> maybe_nullable "bool"
-  | Some Array -> maybe_nullable (process_array_type ~ancestors schema |> String.concat " ")
-  | Some Object -> process_object_type ~ancestors schema
+  | Some Array -> maybe_nullable (process_array_type state ~ancestors schema |> String.concat " ")
+  | Some Object -> process_object_type state ~ancestors schema
   | None ->
     (* fallback to untyped if schema type is not defined *)
-    maybe_nullable "json"
+    Printf.ksprintf maybe_nullable "json (* %s *)" (String.concat "/" (List.rev ancestors))
```

Comment:
Not sure about ancestors lists here, for example:
```diff
   type dynamicConfigValue = {
     ~id  <ocaml default="\"\"">: string;
-    ?value : json option;
+    ?value : json (* DynamicConfigValue/value *) option;
   } 
```
Looks like the comment is a bit redundant here. Do you have an example in mind which would illustrate the reason of having this comment?

Reply:
I've improved the comments but in that case, there is nothing in the JSON to display for `"value"` :)

```json
"DynamicConfigValue": {
  "type": "object",
  "required": [
    "id"
  ],
  "properties": {
    "id": {
      "type": "string",
      "default": ""
    },
    "value": {}
  },
  "$schema": "http://json-schema.org/draft-04/schema#"
},
```